src/gc/windows/gcenv.windows.cpp

```diff
@@ -604,8 +779,132 @@ bool GCToOSInterface::GetWriteWatch(bool resetState, void* address, size_t size,
 //  Size of the cache
 size_t GCToOSInterface::GetCacheSizePerLogicalCpu(bool trueSize)
 {
-    // TODO(segilles) processor detection (see src/vm/util.cpp:1935)
-    return 0;
+    static size_t maxSize;
+    static size_t maxTrueSize;
+
+    if (maxSize)
+    {
+        // maxSize and maxTrueSize cached
+        if (trueSize)
+        {
+            return maxTrueSize;
+        }
+        else
+        {
+            return maxSize;
+        }
+    }
+
+#if defined(_TARGET_AMD64_) || defined (_TARGET_X86_)
```

Comment:
Nit: The GC should not need `_TARGET_XXX_` ifdefs because of it is never cross-compiled. All `_TARGET_XXX_` can be changed to just `_XXX_`.